src/jackdaw/serdes/resolver.clj

```diff
@@ -54,12 +54,16 @@
       (let [schema (cond
                      schema-filename (load-schema serde-config)
                      schema          schema)
-            serde-fn (find-serde-var serde-config)]
-        (case serde-keyword
-          :jackdaw.serdes.avro.confluent/serde
-          (if-not (s/valid? :jackdaw.serde/confluent-avro-serde serde-config)
+            serde-fn (find-serde-var serde-config)
+            spec-mapping {:jackdaw.serdes.avro.confluent/serde :jackdaw.serde/confluent-avro-serde
+                          :jackdaw.serdes.avro.confluent/reader-serde  :jackdaw.serde/confluent-avro-reader-serde}
+            spec-key (spec-mapping serde-keyword)]
+        (case spec-key
+          (:jackdaw.serde/confluent-avro-serde
+           :jackdaw.serde/confluent-avro-reader-serde)
+          (if-not (s/valid? spec-key serde-config)
             (throw (ex-info "Invalid serde config."
-                            (s/explain-data :jackdaw.serde/confluent-avro-serde serde-config)))
+                            (s/explain-data spec-key serde-config)))
```

Comment:
The serde-resolver requires a schema. Can we back out these changes?

Reply:
it's only the serialization side that uses it, and the current kafka-productionised code looks up everything using the resolver

An alternate approach would be to allow the resolver (or something with a similar API) to return a wrapper with only the `deserializer` field.

The current approach is compatible with the existing code paths because the expected deserializer field is present and the resolver adds the expected conversions on read, I'm open to other options but would like to limit the scope of the API change. What about creating a wrapper where the `deserializer` works via the schema registry and the `serializer` simply throws an informative error?